pkg/stores/queryhelper/safesplit.go

```diff
@@ -0,0 +1,16 @@
+package queryhelper
+
+import "strings"
+
+// SafeSplit breaks a regular "a.b.c" path to the string slice ["a", "b", "c"]
+// but if the final part is in square brakets, like "metadata.labels[cattleprod.io/moo]",
+// it returns ["metadata", "labels", "cattleprod.io/moo"
+func SafeSplit(fieldPath string) []string {
+	squareBracketLocation := strings.Index(fieldPath, "[")
+	if squareBracketLocation == -1 {
+		return strings.Split(fieldPath, ".")
+	}
+	s := strings.Split(fieldPath[0:squareBracketLocation], ".")
+	s = append(s, fieldPath[squareBracketLocation+1:len(fieldPath)-1])
```

Comment:
Does the case where `string = "a.b[c"` need any special treatment? (no ending square bracket)

How about `a[[`?

Reply:
Those will just result in failures to match data. I don't see a need to do error-checking here.